src/Analyzer/Pass/Expression/Casts.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace PHPSA\Analyzer\Pass\Expression;
+
+use PhpParser\Node\Expr;
+use PHPSA\Analyzer\Pass\AnalyzerPassInterface;
+use PHPSA\Context;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use PHPSA\CompiledExpression;
+
+class Casts implements AnalyzerPassInterface
+{
+    /**
+     * @param Expr $expr
+     * @param Context $context
+     * @return bool
+     */
+    public function pass(Expr $expr, Context $context)
+    {
+
+        $castType = CompiledExpression::UNKNOWN;
+
+        switch (get_class($expr)) {
+            case Expr\Cast\Array_::class:
+                $castType = CompiledExpression::ARR;
+                break;
+            case Expr\Cast\Bool_::class:
+                $castType = CompiledExpression::BOOLEAN;
+                break;
+            case Expr\Cast\Int_::class:
+                $castType = CompiledExpression::INTEGER;
+                break;
+            case Expr\Cast\Double::class:
+                $castType = CompiledExpression::DOUBLE;
+                break;
+            case Expr\Cast\Object_::class:
+                $castType = CompiledExpression::OBJECT;
+                break;
+            case Expr\Cast\String_::class:
+                $castType = CompiledExpression::STRING;
+                break;
+        }
+
+        $compiledExpression = $context->getExpressionCompiler()->compile($expr->expr);
+        $ExprType = $compiledExpression->getType();
```

Comment:
Camel-case style please, first char must be in lower `$ExprType ` -> `$exprType `
Thanks